### PR TITLE
[AIRFLOW-624] [AIRFLOW-784] [AIRFLOW-785] [AIRFLOW-777] Misc. fixes

### PR DESCRIPTION
**Pull request by gsakkis**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-624
- https://issues.apache.org/jira/browse/AIRFLOW-784
- https://issues.apache.org/jira/browse/AIRFLOW-785
- https://issues.apache.org/jira/browse/AIRFLOW-777

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def do_setup():
             'flask-login==0.2.11',
             'flask-swagger==0.2.13',
             'flask-wtf==0.12',
-            'funcsigs>=1.0.2, <1.1',
```
Comment (bolkedebruin) [inline setup.py:208]: You are setting it lower than we used before.

File: airflow/jobs.py [hunk 2]
```diff
@@ -919,13 +919,7 @@ def _change_state_for_tis_without_dagrun(self,
...
                                          dag_runs))
 
-            dag_is_running = True
-            for dag_run in dag_runs:
-                if dag_run.state == State.RUNNING:
-                    dag_is_running = True
-                    break
-
-            if not dag_is_running:
+            if not any(dag_run.state == State.RUNNING for dag_run in dag_runs):
```
Comment (bolkedebruin) [inline airflow/jobs.py:922]: Good catch

**Comment by bolkedebruin** [comment]
Nice fixes! Small nits / questions. Can you make sure to adhere to the "title" guidelines (mostly your cgroups commit messages is too long).

File: setup.py (hunk 1 above)
Comment (gsakkis) [reply setup.py:208]: Indeed; please check https://issues.apache.org/jira/browse/AIRFLOW-784 why.

Comment (bolkedebruin) [reply setup.py:208]: Mmm this ties it to the older version which would downgrade it at some other places. I'm not sure why we were dependent on >= 1.0.2, so I am not too happy by 'just' downgrading. We could also require a higher version of setup tools.

**Comment by bolkedebruin** [comment]
I think you are missing a new line between the title of AIRFLOW0785 and its body. Github shows "..." which means it still exceeds to maximum amount of columns.

**Comment by bolkedebruin** [comment]
Would be nice btw to have a test for AIRFLOW-777 (i.e. covering the function)

File: setup.py (hunk 1 above)
Comment (gsakkis) [reply setup.py:208]: It's ultimately an issue with funcsigs, I'll open an issue for it.

> We could also require a higher version of setup tools.

Afaict the setuptools requirement was what introduced the funcsigs bug for me; doing the same in Airflow would spread the issue more.

**Comment by gsakkis** [comment]
There is a new line, I think it's because above the 50 characters limit (59 to be precise). Feel free to suggest an alternative shorter subject but IMHO I'd rather pick a descriptive title than chop it off to to whatever arbitrarily small limit a particular git host uses for display.

**Pull request merged**